tests/snappi_tests/multidut/pfc/files/m2o_fluctuating_lossless_helper.py

```diff
@@ -0,0 +1,343 @@
+import logging                                                                          # noqa: F401
+from tests.common.helpers.assertions import pytest_assert, pytest_require               # noqa: F401
+from tests.common.fixtures.conn_graph_facts import conn_graph_facts, fanout_graph_facts  # noqa: F401
+from tests.common.snappi_tests.snappi_helpers import get_dut_port_id                     # noqa: F401
+from tests.common.snappi_tests.common_helpers import pfc_class_enable_vector, stop_pfcwd, \
+    disable_packet_aging, sec_to_nanosec                                                # noqa: F401
+from tests.common.snappi_tests.port import select_ports                                 # noqa: F401
+from tests.common.snappi_tests.snappi_test_params import SnappiTestParams
+from tests.common.snappi_tests.traffic_generation import run_traffic, \
+     setup_base_traffic_config, verify_m2o_oversubscribtion_results
+logger = logging.getLogger(__name__)
+
+PAUSE_FLOW_NAME = 'Pause Storm'
+TEST_FLOW_NAME = 'Test Flow'
+TEST_FLOW_AGGR_RATE_PERCENT = [20, 10]
+BG_FLOW_NAME = 'Background Flow'
+BG_FLOW_AGGR_RATE_PERCENT = [20, 20]
+DATA_PKT_SIZE = 1024
+DATA_FLOW_DURATION_SEC = 20
+DATA_FLOW_DELAY_SEC = 10
+SNAPPI_POLL_DELAY_SEC = 2
+TOLERANCE_THRESHOLD = 0.1
+
+
```

Comment:
please add comments on how the test is done and expected results.

Reply:
The background traffic will include four lossy traffic streams, with any priorities 0..2 and 5..6, each having 20% bandwidth for a total of 80% of the port line rate.The test data traffic will include two lossless traffic flows, with the SONiC default lossless priorities of 3 and 4. Each of lossless traffic flows will be shaped to have line rate of 20% and 10%, so that there are periods where both lossless flows contribute a bandwidth of 30%. Start the background and test traffic from IXIA Tx ports 1 & 2. The combined bandwidth of all the streams going to the egress port will be 110% (over-subscription). After a few seconds, measure the received bandwidth per flow on the IXIA Rx port 3: each of the four lossy flows should have an allocation of 18% of the total egress port bandwidth. Lossless flows should have an allocation of 18% and 10% of the total egress port bandwidth. There should be PFC pause messages being received at the IXIA transmit port(s) causing the over-subscription